cartoframes/client/data_obs_client.py

```diff
@@ -0,0 +1,692 @@
+            '    numer_geomref_colname text, numer_id text,',
+            '    numer_name text, numer_t_description text,',
+            '    numer_tablename text, numer_timespan text,',
+            '    numer_type text, score numeric, score_rank numeric,',
+            '    score_rownum numeric, suggested_name text,',
+            '    target_area text, target_geoms text, timespan_rank numeric,',
+            '    timespan_rownum numeric)',
+            '{quantiles}', )).format(
+                boundary=boundary,
+                numers=numers,
+                quantiles=quantiles).strip()
+        self._debug_print(query=query)
+        return self._fetch(query, decode_geom=True).dataframe
+
+    def augment(self, table_name, metadata, persist_as=None, how='the_geom'):
+        """Get an augmented CARTO dataset with `Data Observatory
+        <https://carto.com/data-observatory>`__ measures. Use
+        `DataObs.discovery
+        <#DataObs.discovery>`__ to search for available
+        measures, or see the full `Data Observatory catalog
+        <https://cartodb.github.io/bigmetadata/index.html>`__. Optionally
+        persist the data as a new table.
+
+        Example:
+            Get a DataFrame with Data Observatory measures based on the
+            geometries in a CARTO table.
+
+            .. code::
+
+                creds = Credentials('user name', 'api key')
+                do = DataObs(creds)
+                median_income = do.discovery(
+                    'transaction_events',
+                    egex='.*median income.*',
+                    time='2011 - 2015')
+                ds = do.augment('transaction_events', median_income)
+
+            Pass in cherry-picked measures from the Data Observatory catalog.
+            The rest of the metadata will be filled in, but it's important to
+            specify the geographic level as this will not show up in the column
+            name.
+
+            .. code::
+
+                median_income = [{'numer_id': 'us.census.acs.B19013001',
+                                  'geom_id': 'us.census.tiger.block_group',
+                                  'numer_timespan': '2011 - 2015'}]
+                ds = do.augment('transaction_events', median_income)
+
+        Args:
+            table_name (str): Name of table on CARTO account that Data
+                Observatory measures are to be added to.
+            metadata (pandas.DataFrame): List of all measures to add to
+                `table_name`. See :py:meth:`DataObs.discovery
+                <cartoframes.data.DataObs.discovery>` outputs
+                for a full list of metadata columns.
+            persist_as (str, optional): Output the results of augmenting
+                `table_name` to `persist_as` as a persistent table on CARTO.
+                Defaults to ``None``, which will not create a table.
+            how (str, optional): **Not fully implemented**. Column name for
```

Comment:
I don't remember why this is not fully implemented. It looks implemented enough, so I think we should remove that message.